backend/src/sudokuboard.py

```diff
@@ -0,0 +1,102 @@
+from collections import deque
+from dataclasses import dataclass
+from functools import singledispatchmethod
+from itertools import islice
+from random import shuffle
+from typing import Final, List, Set
+
+
+@dataclass(eq=True, frozen=True)
+class Coord:
+    i: int = 0
+    j: int = 0
+
+
+@dataclass
+class SudokuCell:
+    pos: Final[Coord]
+    value: int = 0
+    SIZE: Final[int] = 3
+
+    def __post_init__(self):
+        self.neighbors: Set[Coord] = set()
+        for n in range(self.SIZE ** 2):
+            self.neighbors.add(Coord(n, self.pos.j))
+            self.neighbors.add(Coord(self.pos.i, n))
+
+        block_coord = self._get_block_coord()
+
+        for n in range(block_coord.i, block_coord.i + self.SIZE):
+            for m in range(block_coord.j, block_coord.j + self.SIZE):
+                self.neighbors.add(Coord(n, m))
+
+        self.neighbors.remove(self.pos)
+
+    def _get_block_coord(self) -> Coord:
+        return Coord(
+            (self.pos.i // self.SIZE) * self.SIZE,
+            (self.pos.j // self.SIZE) * self.SIZE,
+        )
+
+
+class SudokuBoard:
+    def __init__(self, SIZE=3):
+        self.SIZE = SIZE
+        self.VALID_VALUES = {i + 1 for i in range(SIZE ** 2)}
+        self.cells = deque()
+        for i in range(SIZE ** 2):
+            for j in range(SIZE ** 2):
+                self.cells.append(SudokuCell(Coord(i, j), SIZE=SIZE))
+        self.fill_cells()
+
+    def reset(self):
+        for cell in self.cells:
+            cell.value = 0
+
+    def fill_cells(self):
+        if not self.do_fill_cells(0):
+            raise Exception("Unable to fill the board")
```

Comment:
Actually there is no reason to handle algorithm failure since it will always succeed - ofc if object SIZE is correct etc..
I've got rid of this check, code is now simplified.